core/deployment/src/main/java/io/quarkus/deployment/annotations/ExecutionTime.java

```diff
@@ -11,5 +11,9 @@ public enum ExecutionTime {
     /**
      * The bytecode is run from a main method
      */
-    RUNTIME_INIT
+    RUNTIME_INIT,
+    /**
+     * The bytecode is run from a warmup method
+     */
+    WARMUP_INIT
```

Comment:
Minor but I would put this *before* `RUNTIME_INIT` because the sort order is then chronological.

Reply:
Isn't it bad practice to change the order of an Enum when adding a new element?  Can effect things like DB mappings and any code that uses a number associated with the enum.

Reply:
Only if the ordinal is semantically tied to the value. But I'd argue that one should use a separate property to hold the number in this case. In this case the ordering is more important, allowing them to be compared with each other.